src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import { PartnershipType, UpdatePartnership } from '../../api';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+interface DialogInput {
+  partnership: Except<UpdatePartnership, 'id'>;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const submit: DialogFormProps<DialogInput>['onSubmit'] = async (input) => {
+    const inputWithId = {
+      partnership: { ...input.partnership, id: partnership.id },
+    };
+    await updatePartnership({
+      variables: { input: inputWithId },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+  };
+
+  const onDeleteClick = async () => {
+    const { id } = partnership;
+    await deletePartnership({
+      variables: { input: id },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+    state.onClose();
+  };
+
+  const initialValues: DialogInput = {
+    partnership: {
+      agreementStatus: partnership.agreementStatus.value,
+      mouStatus: partnership.mouStatus.value,
+      types: partnership.types.value,
+    },
+  };
+
+  return (
+    <DialogForm
+      onSubmit={submit}
```

Comment:
Take a look at `EditOrganization` and how I passed the generic to the component and put onSubmit and initialValues inline so I didn't have to explicitly type those variables. 
https://github.com/SeedCompany/cord-field/blob/a42a18dbe55ddf3f5933162662b0ca3e863f5587/src/scenes/Organizations/Edit/EditOrganization.tsx#L23
https://github.com/SeedCompany/cord-field/blob/a42a18dbe55ddf3f5933162662b0ca3e863f5587/src/scenes/Organizations/Edit/EditOrganization.tsx#L31-L41

Reply:
Changed